dynawo/sources/Modeler/DataInterface/DYNDataInterfaceFactory.cpp

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNStateVariable.cpp
+ *
+ * @brief State variable description : implementation file
+ *
+ */
+
+
+#include "DYNDataInterfaceFactory.h"
+#include "DYNDataInterfaceIIDM.h"
+
+using std::string;
+using boost::shared_ptr;
+
+namespace DYN {
+shared_ptr<DataInterface>
+DataInterfaceFactory::build(dataInterfaceType_t type, const string& filepath) {
+  switch(type) {
```

Comment:
I think a switch will not even be necessary and we will use a macro ifndef CXX11_ENABLED because we will not have the two implementations available at the same time. It would be useful if we had something other than iidm

Reply:
Yes, I agree we will use the macro to separate the two libIIDM from each others, but I wanted to have a clean implementation for CGMES or NF for example. Maybe in the commit which wil integrate PowSybl lib IIDM we could rename this DATAINTERFACE_IIDM1 into DATAINTERFACE_IIDM for example

Reply:
I agree with this